### PR TITLE
efficiency tweaks WIP

### PR DESCRIPTION
**Pull request by tduehr**
## Fixes Issue #

partial fix for jeremylong/dependency-check-gradle#100

## Description of Change

Caches some intermediate results and updates loops to more efficient Java 8 Streams and Lambdas.  Lays some of the groundwork to reduce duplicated effort within a single job (currently theis mostly affects Aggregate). More work is required to reduce effort across multiple subprojects for analyze to realize efficiency gains.

## Have test cases been added to cover the new functionality?

N/A

### CODE REVIEW

**Comment by jeremylong** [comment]
Thanks - I'll review this (hopefully) soon. I am in heads down development of 5.0.0 (see https://github.com/jeremylong/DependencyCheck/tree/jsonDataFeeds). There will be a ton of changes beyond what is in that branch.

**Comment by tduehr** [comment]
Is there another branch you're working from for that stuff? It's probably just easier to work from the dev branch...

**Comment by jeremylong** [comment]
@tduehr I am following a more traditional git flow process. The only difference is that the feature branch I pointed to above is going to take a while to finish and touches a lot of the project.  I know I'm going to have some interesting merge conflicts when I rebase the jsonDataFeeds branch.

File: core/pom.xml [hunk 1]
```diff
@@ -142,6 +142,10 @@ Copyright (c) 2012 Jeremy Long. All Rights Reserved.
         </plugins>
     </build>
     <dependencies>
+        <dependency>
```
Comment (jeremylong) [inline core/pom.xml:145]: In a quick review of the code I don't see jackson-databind being used anywhere?

**Comment by tduehr** [comment]
I think that is from trying to debug an error I was getting compiling in jdk 11. I’ll remove it.

**Comment by jeremylong** [comment]
I rebased this PR locally to resolve the issue with animal sniffer (see https://github.com/jeremylong/DependencyCheck/pull/1638). There are several test case failures due to these changes.

**Pull request closed without merging**